packages/language-server/src/xml-view-diagnostics.ts

```diff
@@ -48,18 +48,16 @@ function validationIssuesToLspDiagnostics(
 
     const issueKind = currIssue.kind;
     switch (issueKind) {
-      case "UnknownEnumValue":
-        return {
-          ...commonDiagnosticPros,
-        };
       case "UseOfDeprecatedClass":
         return {
           ...commonDiagnosticPros,
           tags: [DiagnosticTag.Deprecated],
         };
-      /* istanbul ignore next - defensive programming */
       default:
-        assertNever(issueKind);
+        // Most issues don't need any special handling
```

Comment:
we lose the exhaustiveness checking, is it not better to explicitly handle every state
and have the compiler fail if we forget to add/handle a special case?
then silently run with partial functionality?

- https://www.typescriptlang.org/docs/handbook/advanced-types.html#exhaustiveness-checking
- https://www.geeksforgeeks.org/scala-sealed-trait/

Reply:
I don't think so in this case. In the completion items it makes sense because a lot of the parameters are LSP-specific and common, but in the validations we get all the information (except very specific things like the tags) from the inner logic. If we have more specific things (like if we decide to change the API to return the xml node instead of ranges) we can change it. But I don't see a need for it currently.

Reply:
Lets sync on this tomorrow

Reply:
Done